tests/Decoders.fs

```diff
@@ -1442,6 +1450,44 @@ Expecting an object with a field named `version` but instead got:
 
                 equal expected actual
 
+
+            testCase "all works" <| fun _ ->
+                let expected = Ok [1; 2; 3]
+
+                let decodeAll = Decode.all [
+                    Decode.succeed 1
+                    Decode.succeed 2
+                    Decode.succeed 3
+                ]
+
+                let actual = Decode.fromString decodeAll "{}"
+
+                equal expected actual
+
+
+            testCase "all succeeds on empty lists" <| fun _ ->
+                let expected = Ok []
+
+                let decodeNone = Decode.all []
+
+                let actual = Decode.fromString decodeNone "{}"
+
+                equal expected actual
+
+
+            testCase "all fails when one decoder fails" <| fun _ ->
+                let msg = "Failed"
```

Comment:
Could you please write the test with another decoder than `Decode.fail` just to make sure the error message is useful to understand what happened?

Reply:
Done